crates/net/eth-wire-types/src/status.rs

```diff
@@ -306,6 +306,47 @@ impl From<Status> for StatusEth69 {
     }
 }
 
+impl From<StatusEth69> for Status {
+    fn from(s: StatusEth69) -> Self {
+        Self {
+            version: s.version,
+            chain: s.chain,
+            // Eth69 doesn't include total_difficulty so we choose a default
+            total_difficulty: U256::default(),
+            blockhash: s.blockhash,
+            genesis: s.genesis,
+            forkid: s.forkid,
+        }
+    }
+}
+
+/// `StatusPayload` can store either the Legacy version (with TD) or the
+/// eth/69 version (omits TD).
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum StatusPayload {
```

Comment:
what we can do for this type is add some helper functions/getters to access internals that we need to check like genesis, then we likely don't need the from conversion above.

we could however convert the from impl into a function and document that this conversion isn't complete (which is fine), that would make it easier to apply those checks in the handshake.

both options are okay.

Reply:
@mattsse, luv the helper functions idea! Implemented in latest commit https://github.com/paradigmxyz/reth/pull/15543/commits/c8a6ee00e1f577a34a4c1689f1832f6291889f34

we still need the `impl From<StatusEth69> for Status` since the `eth_handshake` needs to return the peer's legacy `Status` https://github.com/paradigmxyz/reth/blob/19f8070565bd4bedc4b06d113b4fbc249410c875/crates/net/eth-wire/src/handshake.rs#L82-L86